### ISSUE TITLE
Upgrade to HTML Unit 2.23.3 or later

### ISSUE THREAD

**Issue by ylu999** [spring-projects/spring-boot#7232]
When running spring-boot-sample-test from one of the samples, we missing dependency class com.gargoylesoftware.htmlunit.html.HtmlPage
It turns out that we need to upgrade htmlunit to 2.23, and problem solved.

**Comment by snicoll** [comment]
Closing in favour of PR #7233

@ylu999 no need to create both an issue and a PR. A PR is enough.

**Comment by philwebb** [comment]
Reopen to add additional property.

**Comment by snicoll** [comment]
Unfortunately, 2.23 is breaking other use cases so I'd like us to reconsider the upgrade.

@ylu999  you can still perfectly work with your use cases with the upcoming Spring Boot `1.5.0.RC1` release by adding the following properties (assuming you are using the `spring-boot-starter-parent` as the parent):

```
<selenium-htmlunit.version>2.23.2</selenium-htmlunit.version>
<htmlunit.version>2.23</htmlunit.version>
```

**Comment by snicoll** [comment]
Waiting for a release including the fix for https://github.com/SeleniumHQ/htmlunit-driver/issues/22

**Comment by wilkinsona** [comment]
The fix we need is available in 2.24
